modelcontextprotocol/tools/assets.py

```diff
@@ -184,3 +190,88 @@ def update_assets(
         error_msg = f"Error updating assets: {str(e)}"
         logger.error(error_msg)
         return {"updated_count": 0, "errors": [error_msg]}
+
+
+def get_asset_history(
+    guid: Optional[str] = None,
+    qualified_name: Optional[str] = None,
+    type_name: Optional[str] = None,
+    size: int = 10,
+    sort_order: str = "DESC",
+) -> Dict[str, Any]:
+    """
+    Get the audit history of an asset by GUID or qualified name.
+
+    Args:
+        guid (Optional[str]): GUID of the asset to get history for.
+            Either guid or qualified_name must be provided.
+        qualified_name (Optional[str]): Qualified name of the asset to get history for.
+            Either guid or qualified_name must be provided.
+        type_name (Optional[str]): Type name of the asset (required when using qualified_name).
+            Examples: "Table", "Column", "DbtModel", "AtlasGlossary"
+        size (int): Number of history entries to return. Defaults to 10.
+        sort_order (str): Sort order for results. "ASC" for oldest first, "DESC" for newest first.
+            Defaults to "DESC".
+
+    Returns:
+        Dict[str, Any]: Dictionary containing:
+            - entityAudits: List of audit entries
+            - count: Number of audit entries returned
+            - totalCount: Total number of audit entries available
+            - errors: List of any errors encountered
+
+    Raises:
+        Exception: If there's an error retrieving the asset history
+    """
+    try:
+        # Validate input parameters
+        validation_error = validate_asset_history_params(
+            guid, qualified_name, type_name, sort_order
+        )
+        if validation_error:
+            logger.error(validation_error)
+            return {
+                "errors": [validation_error],
+                "entityAudits": [],
+                "count": 0,
+                "totalCount": 0,
+            }
+
+        logger.info(
+            f"Retrieving asset history with parameters: guid={guid}, qualified_name={qualified_name}, size={size}"
+        )
+
+        # Get Atlan client
+        client = get_atlan_client()
+
+        # Create sort item
+        sort_item = create_sort_item(sort_order)
+
+        # Create and execute audit search request
+        request = create_audit_search_request(
+            guid, qualified_name, type_name, size, sort_item
+        )
+        response = client.audit.search(criteria=request, bulk=False)
```

Comment:
We should catch exceptions and return them as errors since you have defined the tool in that way